pkg/apis/management.cattle.io/v3/authz_types.go

```diff
@@ -135,19 +135,37 @@ func (p *ProjectSpec) ObjClusterName() string {
 }
 
 // +genclient
-// +kubebuilder:skipversion
 // +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// GlobalRole defines rules that can be applied to the local cluster and or every downstream cluster.
 type GlobalRole struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	DisplayName    string              `json:"displayName,omitempty" norman:"required"`
-	Description    string              `json:"description"`
-	Rules          []rbacv1.PolicyRule `json:"rules,omitempty"`
-	NewUserDefault bool                `json:"newUserDefault,omitempty" norman:"required"`
-	Builtin        bool                `json:"builtin" norman:"nocreate,noupdate"`
+	// DisplayName is the human-readable name displayed in the UI for this resource.
+	// +optional
+	DisplayName string `json:"displayName,omitempty" norman:"required"`
+
+	// Description holds text that describes the resource.
+	// +optional
+	Description string `json:"description"`
```

Comment:
This is marked as optional but there's no `omitempty` tag (unlike the displayName which is of the same type) - can you elaborate on why?

Reply:
Just a simple 🐛